markets/idxprov/mesh.go

```diff
@@ -21,16 +25,25 @@ type Libp2pMeshCreator struct {
 }
 
 func (mc Libp2pMeshCreator) Connect(ctx context.Context) error {
-	addrs, err := mc.fullnodeApi.NetAddrsListen(ctx)
+	faddrs, err := mc.fullnodeApi.NetAddrsListen(ctx)
 	if err != nil {
 		return err
 	}
 
-	if err := mc.idxProvHost.Connect(ctx, addrs); err != nil {
+	// if we already have a connection to the full node, there's nothing to do here
+	if mc.idxProvHost.Network().Connectedness(faddrs.ID) == network.Connected {
+		return nil
```

Comment:
is there a race here where we're connected, but don't protect?
an extra protect shouldn't hurt since it should be idempotent if done with the same tag.

Reply:
Addressed this.